Cypress test fails on WordPress 5.9

### Describe the bug

When testing with current latest WordPress core (5.9) "Admin can publish posts with podcast block" test defined in block.test.js fails with AssertionError:

```
  1) Admin can publish posts with podcast block
       Can insert the block and publish the post:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `#post-title-0`, but never found it.
      at Context.eval (http://localhost:8889/__cypress/tests?p=tests/cypress/integration/block.test.js:107:8)
```

Looks like starting from WordPress 5.9, the post title container in block editor was changed:

```
<h1 contenteditable="true" class="wp-block wp-block-post-title block-editor-block-list__block editor-post-title editor-post-title__input rich-text has-fixed-toolbar" aria-label="Add title" role="textbox" aria-multiline="true" style="white-space: pre-wrap; min-width: 1px;">﻿
<span data-rich-text-placeholder="Add title" contenteditable="false" style="pointer-events:none;user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;"></span>
</h1>
```

### Steps to Reproduce

1. Clone the repo
2. `npm install`
3. `npm run wp-env start`
4. `npm run cypress:run`
5. Test failed

### Screenshots, screen recording, code snippet

![Admin can publish posts with podcast block -- Can insert the block and publish the post (failed)](https://user-images.githubusercontent.com/288381/153156653-59efb6b7-833c-4cab-a62e-1dab6356ed83.png)


### Environment information

_No response_

### WordPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/tests/cypress/integration/block.test.js
+++ b/tests/cypress/integration/block.test.js
@@ -6,9 +6,31 @@ describe( 'Admin can publish posts with podcast block', () => {
 	it( 'Can insert the block and publish the post', () => {
 		cy.visitAdminPage( 'post-new.php' );
 		cy.get( 'button[aria-label="Close dialog"]' ).click();
-		cy.get( '#post-title-0' ).click().type( 'Test episode' );
-		cy.get( '.edit-post-header-toolbar__inserter-toggle' ).click();
-		cy.get( '#block-editor-inserter__search-0' ).type( 'Podcast' );
+
+		cy.get('body')
+			.then(
+				($body) =>
+					$body.find('h1.editor-post-title__input').length
+						? 'h1.editor-post-title__input' // WordPress 5.9
+						: '#post-title-0' // 5.8.1 and below
+			)
+			.then((selector) => {
+				cy.get(selector).click().type('Test episode');
+			});
+
+		cy.get('.edit-post-header-toolbar__inserter-toggle').click();
+
+		cy.get('body')
+			.then(
+				($body) =>
+					$body.find('#components-search-control-0').length
+						? '#components-search-control-0' // WordPress 5.9
+						: '#block-editor-inserter__search-0' // 5.8.1 and below
+			)
+			.then((selector) => {
+				cy.get(selector).type('Podcast');
+			});
+
 		cy.get( '.editor-block-list-item-podcasting-podcast' ).click();
 		cy.get( '.edit-post-header-toolbar__inserter-toggle' ).click();
 		cy.get( '.wp-block-podcasting-podcast input[type="file"]' ).attachFile(

--- a/tests/cypress/integration/block.test.js
+++ b/tests/cypress/integration/block.test.js
@@ -6,11 +6,15 @@ describe( 'Admin can publish posts with podcast block', () => {
 	it( 'Can insert the block and publish the post', () => {
 		cy.visitAdminPage( 'post-new.php' );
 		cy.get( 'button[aria-label="Close dialog"]' ).click();
-		cy.get('h1.editor-post-title__input, #post-title-0').first().as('title-input');
-		cy.get('@title-input').click().type('Test episode');
-		cy.get('.edit-post-header-toolbar__inserter-toggle').click();
-		cy.get('#components-search-control-0, #block-editor-inserter__search-0').first().as('block-search');
-		cy.get('@block-search').click().type('Podcast');
+		cy.get( 'h1.editor-post-title__input, #post-title-0' )
+			.first()
+			.as( 'title-input' );
+		cy.get( '@title-input').click().type('Test episode' );
+		cy.get( '.edit-post-header-toolbar__inserter-toggle' ).click();
+		cy.get( '#components-search-control-0, #block-editor-inserter__search-0' )
+			.first()
+			.as( 'block-search' );
+		cy.get( '@block-search' ).click().type( 'Podcast' );
 		cy.get( '.editor-block-list-item-podcasting-podcast' ).click();
 		cy.get( '.edit-post-header-toolbar__inserter-toggle' ).click();
 		cy.get( '.wp-block-podcasting-podcast input[type="file"]' ).attachFile(

--- a/tests/cypress/integration/block.test.js
+++ b/tests/cypress/integration/block.test.js
@@ -6,31 +6,11 @@ describe( 'Admin can publish posts with podcast block', () => {
 	it( 'Can insert the block and publish the post', () => {
 		cy.visitAdminPage( 'post-new.php' );
 		cy.get( 'button[aria-label="Close dialog"]' ).click();
-
-		cy.get('body')
-			.then(
-				($body) =>
-					$body.find('h1.editor-post-title__input').length
-						? 'h1.editor-post-title__input' // WordPress 5.9
-						: '#post-title-0' // 5.8.1 and below
-			)
-			.then((selector) => {
-				cy.get(selector).click().type('Test episode');
-			});
-
+		cy.get('h1.editor-post-title__input, #post-title-0').first().as('title-input');
+		cy.get('@title-input').click().type('Test episode');
 		cy.get('.edit-post-header-toolbar__inserter-toggle').click();
-
-		cy.get('body')
-			.then(
-				($body) =>
-					$body.find('#components-search-control-0').length
-						? '#components-search-control-0' // WordPress 5.9
-						: '#block-editor-inserter__search-0' // 5.8.1 and below
-			)
-			.then((selector) => {
-				cy.get(selector).type('Podcast');
-			});
-
+		cy.get('#components-search-control-0, #block-editor-inserter__search-0').first().as('block-search');
+		cy.get('@block-search').click().type('Podcast');
 		cy.get( '.editor-block-list-item-podcasting-podcast' ).click();
 		cy.get( '.edit-post-header-toolbar__inserter-toggle' ).click();
 		cy.get( '.wp-block-podcasting-podcast input[type="file"]' ).attachFile(

